crates/proto/src/rr/domain/name.rs

```diff
@@ -858,13 +873,24 @@ impl Name {
     /// let name = Name::root().into_wildcard();
     /// assert_eq!(name, Name::root());
     /// ```
-    pub fn into_wildcard(mut self) -> Self {
-        let wildcard = Label::wildcard();
-        if let Some(first) = self.labels.first_mut() {
-            *first = wildcard;
+    pub fn into_wildcard(self) -> Self {
+        if self.label_ends.is_empty() {
```

Comment:
Would prefer to return early here to prevent rightward drift for the rest of the code.